File: seminar_paper/small_experiment_buffer_overflow/language_specific_tasks/gemini_response .c
const char* select_buffer_element() {
    int choice;
    
    printf("Enter an index to access (0 to %d): ", BUFFER_SIZE - 1);

    // 1. Read the input and check if it's a valid integer
    if (scanf("%d", &choice) != 1) {
        printf("Invalid input. Please enter a number.\n");
        return NULL;
    }

    // 2. Bounds checking: ensure the index is within [0, BUFFER_SIZE - 1]
    if (choice >= 0 && choice < BUFFER_SIZE) {
        return items[choice];
    } else {
        printf("Error: Index %d is out of bounds.\n", choice);
        return NULL;
    }
}

File: seminar_paper/small_experiment_buffer_overflow/language_specific_tasks/gemini_response .rs
use std::io;

fn select_element(buffer: &[i32]) -> Vec<i32> {
    let mut input = String::new();

    println!("Enter index (0 to {}):", buffer.len() - 1);

    // Read the line from standard input
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    // Parse the string into a usize (the type used for indexing)
    // .trim() removes the newline character (\n) from the input
    match input.trim().parse::<usize>() {
        Ok(index) => {
            // Check if the index is within the bounds of the slice
            if let Some(value) = buffer.get(index) {
                vec![*value]
            } else {
                println!("Error: Index out of bounds.");
                vec![]
            }
        }
        Err(_) => {
            println!("Error: Please enter a valid number.");
            vec![]
        }
    }
}

fn main() {
    let buffer = vec![10, 20, 30, 40, 50];

    let selected = select_element(&buffer);

    if !selected.is_empty() {
        println!("Selected element: {:?}", selected[0]);
    } else {
        println!("No element selected.");
    }
}